cypress/integration/user_account/user_account_test.js

```diff
@@ -0,0 +1,146 @@
+            cy.get('[type="checkbox"]').first().should("not.be.checked");
+        })
+    })
+
+    it("can generate API access key", function(){
+        cy.loginToUserAccount("test_user_account@mail.com","password");
+        cy.visit("/biotestmine/profile");
+        cy.get('.profile-page > ').eq(1).as("generateAPI").within(() => {
+            cy.contains("Generate a new API key").click();
+            cy.get(".success").should("include.text","New API key successfully generated.");
+        })
+    })
+
+    it("can add tags to list", function(){
+        cy.loginToUserAccount("test_user_account@mail.com","password");
+        cy.createGeneList("SODB, GBP, GST, CDPK1");
+        cy.contains("Lists").click();
+        cy.url().should("include","/lists");
+        cy.get('.lists-item').its("length").should('be.gt',0);
+
+        cy.get(".lists-item").first().within(() => {
+            cy.get("button").find('svg[class="icon icon-list-more"]')
+            .should("be.visible")
+            .click();
+
+            cy.get("button").find('svg[class="icon icon-list-edit"]')
+            .should("be.visible")
+            .first()
+            .click();
+        });
+
+        cy.get(".modal-body").within(()=>{
+            cy.contains("Tags").parent().find("input").type("Algae genes{enter}",{delay:100});
+        })
+        cy.get('.modal-footer').within(()=>{
+            cy.contains("Save").click();
+        })
+        cy.get(".lists-table").within(()=>{
+            cy.get(".tag").should("contain","Algae genes");
+        })
+    })
+
+    it("can add lists to folders", function(){
+        cy.loginToUserAccount("test_user_account@mail.com","password");
+        cy.contains("Lists").click();
+        cy.url().should("include","/lists");
+        cy.get('.lists-item').its("length").should('be.gt',0);
+
+        cy.get('.lists-headers').within(() => {
+            cy.get('[type="checkbox"]').check();
+        })
+        cy.get('.bottom-controls').within(() => {
+            cy.contains("Move all").click();
+        })
+        cy.get(".css-1wy0on6").click().type("List of common genes{enter}",{delay:100},{force:true});
+        cy.get(".modal-footer").within(() => {
+            cy.contains("Move list(s)").click();
+        })
+        cy.get('.list-title').should("contain","List of common genes");
+    })
+
+    it("can filter lists by tags", function(){
+        cy.loginToUserAccount("test_user_account@mail.com","password");
+        cy.createProteinList("Q8ID23_PLAF7, Q6LFN1");
+        cy.get('.lists-item').its("length").should('be.gt',0);
+
+        cy.contains("Tags").parent().find(".icon-selection").click();
+        cy.get(".dropdown").find("li").filter(':contains("Algae genes")').click();
+        cy.get('.lists-item').should('have.length',1);
+        cy.get('.lists-item').find(".list-title").should("include.text","List of common genes");
+    })
+
+    it("can build and save a query", function(){
```

Comment:
After saving the query, can you add another `cy.visit("/biotestmine/querybuilder");` (to make sure the session gets cleared - you might have to login again) and then assert that the saved query is still there?